common/lib/xmodule/xmodule/js/src/capa/display.js

```diff
@@ -377,406 +441,461 @@
+                            fileTooLarge = true;
+                            maxSize = maxFileSize / (1000 * 1000);
+                            errors.push(edx.StringUtils.interpolate(
+                                gettext('Your file {filename} is too large (max size: {maxSize}MB).'), {
+                                    filename: file.name,
+                                    maxSize: maxSize
+                                }
+                            ));
                         }
                         fd.append(element.id, file);
                     }
                     if (element.files.length === 0) {
-                        file_not_selected = true;
+                        fileNotSelected = true;
                         fd.append(element.id, '');
                     }
-                    if (required_files.length !== 0) {
-                        required_files_not_submitted = true;
-                        return errors.push("You did not submit the required files: " + required_files + ".");
+                    if (requiredFiles.length !== 0) {
+                        requiredFilesNotSubmitted = true;
+                        errors.push(edx.StringUtils.interpolate(
+                            gettext('You did not submit the required files: {requiredFiles}.'), {
+                                requiredFiles: requiredFiles
+                            }
+                        ));
                     }
                 } else {
-                    return fd.append(element.id, element.value);
+                    fd.append(element.id, element.value);
                 }
             });
-            if (file_not_selected) {
-                errors.push('You did not select any files to submit');
+            if (fileNotSelected) {
+                errors.push(gettext('You did not select any files to submit.'));
             }
-            error_html = '<ul>\n';
-            for (_i = 0, _len = errors.length; _i < _len; _i++) {
-                error = errors[_i];
-                error_html += '<li>' + error + '</li>\n';
+            errorHtml = '<ul>\n';
+            for (i = 0, len = errors.length; i < len; i++) {
+                error = errors[i];
+                errorHtml += '<li>' + error + '</li>\n';
             }
-            error_html += '</ul>';
-            this.gentle_alert(error_html);
-            abort_submission = file_too_large || file_not_selected || unallowed_file_submitted || required_files_not_submitted;
-            if (abort_submission) {
-                window.clearTimeout(timeout_id);
+            errorHtml += '</ul>';
+            this.gentle_alert(errorHtml);
+            abortSubmission = fileTooLarge || fileNotSelected || unallowedFileSubmitted || requiredFilesNotSubmitted;
+            if (abortSubmission) {
+                window.clearTimeout(timeoutId);
                 this.enableSubmitButton(true);
-                return;
```

Comment:
Yes, I spent a lot of time trying to understand if a return was actually needed (when eslint was complaining about it). But you are quite correct-- I got this one wrong! Thanks for catching it!

I will put an "else" around the code below this, to avoid the return but get the correct behavior.